visualizer/src/App.cpp

```diff
@@ -47,43 +47,49 @@ void App::init()
 //Basically just focusing on the x,y plane as of now. 
 void App::update()
 {
-    angle += 0.1f; 
     //test_mesh.position.x = radius * cosf(angle); 
     //test_mesh.position.y = radius * sinf(angle);
-
     Keyboard::poll();
     
     if(Keyboard::isDown(SDL_SCANCODE_R) || !start)
     {
         //Whenever the key R is pressed or the program started up, the camera will look at the default view 
-        Camera::lookAt(glm::vec3(test_mesh.position.x, test_mesh.position.y, 20), test_mesh.position + glm::vec3(0, 10, 0));
+        Camera::lookAt(glm::vec3(test_mesh.position.x, test_mesh.position.y, radius), test_mesh.position + glm::vec3(0, 10, 0));
         start = true; 
+
+        x = y = z = 0.0f; 
     }
     else if (Keyboard::isDown(SDL_SCANCODE_LEFT))
     {
-        x += 0.1f; 
-        z += 0.1f; 
+            prev_view = radius - (radius * cosf(z)); 
+
+        x += angle; 
+        z += angle;
         //Note the camera should move on the x-z plane, the angle here for x and z are incrementing to move left 
-        Camera::lookAt(test_mesh.position + glm::vec3(radius*sin(x), 0, radius*cos(z)), test_mesh.position + glm::vec3(0, 10, 0));
+        Camera::lookAt(test_mesh.position + glm::vec3(radius*sinf(x), 0, radius*cosf(z)), test_mesh.position + glm::vec3(0, 10, 0));
+        //Camera::lookAt(test_mesh.position + glm::vec3(prev_view*sin(x), radius * sinf(y), prev_view*cosf(z)) , test_mesh.position + glm::vec3(0, 10, 0));
     }
     else if (Keyboard::isDown(SDL_SCANCODE_RIGHT))
     {
-        x -= 0.1f; 
-        z -= 0.1f; 
+        //prev_view = radius - (radius * cosf(z)); Ignore; will be helpful tho for the quaternion camera stuff 
```

Comment:
This doesn't make much sense atm, but I am trying to orbit around an object at any position. So imagine I press the 'up' key and then I want to move right or left, at that view, the camera only has to travel through a smaller circumference than if it was orbiting the center of sphere.....................yeah I'm not sure if this makes sense.